### PR TITLE
Azure_hyperv transformer 

### PR DESCRIPTION
**Pull request by SRIKKANTH**
It sets up required tools to make an Azure Windows VM into L1VH.

### CODE REVIEW

File: lisa/transformers/azure_hyperv.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+@dataclass_json
+@dataclass
+class AzureHypervTransformerSchema(schema.Transformer):
+    # SSH connection information to the node
+    connection: Optional[schema.RemoteNode] = field(
+        default=None, metadata=field_metadata(required=True)
+    )
+
+
+class AzureHypervTransformer(Transformer):
```
Comment (squirrelsc) [inline lisa/transformers/azure_hyperv.py]: The name is not clear enough on its purpose. How about `HyperVPreparationTransformer`?

Comment (SRIKKANTH) [reply lisa/transformers/azure_hyperv.py]: Renamed the transformer to "hyperv_preparation"

File: lisa/transformers/hyperv_preparation.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+from lisa import schema
+from lisa.node import quick_connect
+from lisa.tools import PowerShell
+from lisa.transformer import Transformer
+from lisa.util import field_metadata
+
+
+@dataclass_json
+@dataclass
+class HyperVPreparationTransformerSchema(schema.Transformer):
```
Comment (squirrelsc) [inline lisa/transformers/hyperv_preparation.py]: Refer to `DisableCloudComponentsTransformerSchema` to reuse `DeploymentTransformerSchema`.

Comment (SRIKKANTH) [reply lisa/transformers/hyperv_preparation.py]: Modified as suggested.

File: lisa/transformers/hyperv_preparation.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    def _output_names(self) -> List[str]:
+        return []
+
+    def _internal_run(self) -> Dict[str, Any]:
+        runbook: HyperVPreparationTransformerSchema = self.runbook
+        assert isinstance(runbook, HyperVPreparationTransformerSchema)
+        assert (
+            runbook.connection
+        ), "'connection' must be defined if not running during deployed phase."
+        node = quick_connect(runbook.connection, runbook.name, parent_logger=self._log)
```
Comment (squirrelsc) [inline lisa/transformers/hyperv_preparation.py]: When reusing the DeploymentTransformer, the node is initialized in DeploymentTransformer. You can use it directly.

File: lisa/transformers/hyperv_preparation.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    This Transformer configures Windows Azure VM as a Hyper-V environment.
+    """
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "hyperv_preparation"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return HyperVPreparationTransformerSchema
```
Comment (squirrelsc) [inline lisa/transformers/hyperv_preparation.py]: It can return `DeploymentTransformerSchema`, no need another schema without new fields.

Comment (SRIKKANTH) [reply lisa/transformers/hyperv_preparation.py]: Modified as suggested.

File: lisa/transformers/hyperv_preparation.py (hunk 3 above)
Comment (SRIKKANTH) [reply lisa/transformers/hyperv_preparation.py]: Modified as suggested.

File: lisa/transformers/hyperv_preparation.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+from lisa.tools import PowerShell
+from lisa.transformers.deployment_transformer import (
+    DeploymentTransformer,
+    DeploymentTransformerSchema,
+)
+
+
+@dataclass_json
+@dataclass
+class HyperVPreparationTransformerSchema(DeploymentTransformerSchema):
```
Comment (squirrelsc) [inline lisa/transformers/hyperv_preparation.py]: Remove the unused schema.

Comment (SRIKKANTH) [reply lisa/transformers/hyperv_preparation.py]: Modified as suggested.

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**